### ISSUE TITLE
Unstable macOS test in nightly

### ISSUE THREAD

**Issue by keyboardDrummer** [dafny-lang/dafny#3107]
https://github.com/dafny-lang/dafny/actions/runs/3538283931/jobs/5938991371
https://github.com/dafny-lang/dafny/actions/runs/3545855482/jobs/5954352951

```
Starting test execution, please wait...
A total of 1 test files matched the specified pattern.
[xUnit.net 00:10:06.34]     dafny0/snapshots/Snapshots7.run.dfy [SKIP]
  Skipped dafny0/snapshots/Snapshots7.run.dfy [1 ms]
[xUnit.net 00:14:21.00]     DafnyTests/TestAttribute.dfy [FAIL]
  Failed DafnyTests/TestAttribute.dfy [6 s]
  Error Message:
   System.Exception : Command returned non-zero exit code (1): OutputCheck --file-to-check /Users/runner/work/dafny/dafny/dafny/Source/IntegrationTests/bin/Debug/net6.0/TestFiles/LitTests/LitTest/DafnyTests/Output/TestAttribute.dfy.tmp TestFiles/LitTests/LitTest/DafnyTests/TestAttribute.dfy
Output:

Error:
ERROR: Could not find a match for Check Directive (TestFiles/LitTests/LitTest/DafnyTests/TestAttribute.dfy:8 Pattern: '.*_module.__default.FailingTest_CheckForFailureForXunit.*')
  Stack Trace:
     at XUnitExtensions.Lit.LitTestCase.Execute(ITestOutputHelper outputHelper) in /Users/runner/work/dafny/dafny/dafny/Source/XUnitExtensions/Lit/LitTestCase.cs:line 94
   at XUnitExtensions.Lit.LitTestCase.Run(String filePath, LitTestConfiguration config, ITestOutputHelper outputHelper) in /Users/runner/work/dafny/dafny/dafny/Source/XUnitExtensions/Lit/LitTestCase.cs:line 55
   at IntegrationTests.LitTests.LitTest(String path) in /Users/runner/work/dafny/dafny/dafny/Source/IntegrationTests/LitTests.cs:line 1[40](https://github.com/dafny-lang/dafny/actions/runs/3538283931/jobs/5938991371#step:20:41)
  Standard Output Messages:
 Executing command: /Users/runner/work/dafny/dafny/unzippedRelease/dafny/dafny /countVerificationErrors:0 /vcsCores:2 /useBaseNameForFileName /compileVerbose:0 /timeLimit:300 /compileVerbose:1 /compile:0 /spillTargetCode:3 /noVerify TestFiles/LitTests/LitTest/DafnyTests/TestAttribute.dfy > /Users/runner/work/dafny/dafny/dafny/Source/IntegrationTests/bin/Debug/net6.0/TestFiles/LitTests/LitTest/DafnyTests/Output/TestAttribute.dfy.tmp
 Executing command: ! dotnet test -v:q -noLogo /Users/runner/work/dafny/dafny/dafny/Source/IntegrationTests/bin/Debug/net6.0/TestFiles/LitTests/LitTest/DafnyTests >> /Users/runner/work/dafny/dafny/dafny/Source/IntegrationTests/bin/Debug/net6.0/TestFiles/LitTests/LitTest/DafnyTests/Output/TestAttribute.dfy.tmp
 Executing command: OutputCheck --file-to-check /Users/runner/work/dafny/dafny/dafny/Source/IntegrationTests/bin/Debug/net6.0/TestFiles/LitTests/LitTest/DafnyTests/Output/TestAttribute.dfy.tmp TestFiles/LitTests/LitTest/DafnyTests/TestAttribute.dfy
 Previous command results:
 Output:
 
 Dafny program verifier did not attempt verification
 TestAttribute.dfy(117,21): Error: Post-conditions on function Identity might be unsatisfied when synthesizing code for method mockUnsafe
 TestAttribute.dfy(207,21): Error: Stubbing fields is not recommended (field value of object e inside method MockField)
 TestAttribute.dfy(218,21): Error: Stubbing fields is not recommended (field value of object e inside method ParametrizedMock)
 Wrote textual form of partial target program to TestAttribute.cs
 
 Error:
 ERROR: Could not find a match for Check Directive (TestFiles/LitTests/LitTest/DafnyTests/TestAttribute.dfy:8 Pattern: '.*_module.__default.FailingTest_CheckForFailureForXunit.*')
 Output:
 MSBUILD : error MSB1003: Specify a project or solution file. The current working directory does not contain a project or solution file.
 
 Error:
 ERROR: Could not find a match for Check Directive (TestFiles/LitTests/LitTest/DafnyTests/TestAttribute.dfy:8 Pattern: '.*_module.__default.FailingTest_CheckForFailureForXunit.*')
```

Also:

https://github.com/dafny-lang/dafny/actions/runs/3545855482/jobs/5954352864

```
Output:
 include "/Users/runner/work/dafny/dafny/dafny/Source/IntegrationTests/bin/Debug/net6.0/TestFiles/LitTests/LitTest/DafnyTestGeneration/Output/TestGeneration.dfy.tmp.dfy"
 module TestGenerationUnitTests {
 import M
 method {:test} test0() {
 var v0 := getFreshMValue();
 v0.v := -39;
 var r0 := M.compareToZero(v0);
 }
 method {:test} test1() {
 var v0 := getFreshMValue();
 v0.v := 39;
 var r0 := M.compareToZero(v0);
 }
 method {:test} test2() {
 var v0 := getFreshMValue();
 v0.v := 0;
 var r0 := M.compareToZero(v0);
 }
 
 method {:synthesize} getFreshMValue() returns (o:M.Value) ensures fresh(o)
 }
 
 Error:
 
 Output:
 
 Dafny program verifier did not attempt verification
 Wrote textual form of target program to TestGeneration.dfy.tmp-tests.cs
```

**Comment by keyboardDrummer** [comment]
Based on https://github.com/dafny-lang/dafny/actions/runs/3548443568/jobs/5959663116, the tests also fail on Ubuntu and Windows. It seems these tests just fail with some 50% likelihood.

**Comment by robin-aws** [comment]
I finally spotted that this is the actual error that isn't expected in this test:

```
MSBUILD : error MSB1003: Specify a project or solution file. The current working directory does not contain a project or solution file.
```

i.e. `dotnet test` is expected to return a non-zero exit code because of failing tests, but it doesn't actually get that far.

I'm also noticing that the behaviour of the `actions/setup-dotnet` action changed slightly on the day that `TestAttribute.dfy` started to be flaky. 

Before, it reported that it was downloading 6.0.403: https://github.com/dafny-lang/dafny/actions/runs/3520982299/jobs/5902396246#step:4:14

After, it started reporting that 6.0.403 was already installed: https://github.com/dafny-lang/dafny/actions/runs/3529784188/jobs/5921112602#step:4:14

In other words, the MacOS GitHub Actions runners started to include the version already, which means we may be using a slightly different dotnet SDK, or at least a slightly different setup. I'm going to see if just tweaking the version to force the original behaviour is a quick fix.


> Based on https://github.com/dafny-lang/dafny/actions/runs/3548443568/jobs/5959663116, the tests also fail on Ubuntu and Windows. It seems these tests fail with some about 50% likelihood.

See https://github.com/dafny-lang/dafny/pull/3113/files#r1032659519, it's a red herring because of adding `// RUN: false` :)
